ARTESCA-1878: Dashboard Network Plane Health

theocerutti opened the pull request:
![screen](https://user-images.githubusercontent.com/44285344/131853743-67246496-bb7b-4fcf-804c-3ffb0ab76a5f.png)
![Screenshot from 2021-09-02 15-38-13](https://user-images.githubusercontent.com/44285344/131853857-ac84a9cf-24c3-4a44-be2a-e0a1af74e92b.png)



Closes: #3511

bert-e commented:
# Hello theocerutti,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @lucieleonard

* @teddyandrieux

* @thomasdanan

* @nicolast

* @chengyanjin

* @jbwatenbergscality

* @gdemonet

* @alexandre-allard-scality

ChengYanJin commented on ui/src/components/HealthItem.js:
```diff
@@ -0,0 +1,166 @@
+          placement="top"
+          overlayStyle={{
+            // em sizing to handle font-size change on large screens
+            width: '20em',
```
In order to handle responsiveness font, we use `rem` instead of `em` :)

ChengYanJin commented on ui/src/translations/fr.json:
```diff
@@ -137,6 +137,8 @@
   "backend_disk": "Backend Disk",
   "metrics": "Métrique",
+  "network": "Réseau",
+  "planes": "Plans",
```
IMO, we don't need to translate the `Planes`.

ChengYanJin commented on ui/src/components/DashboardPlane.js:
```diff
@@ -0,0 +1,62 @@
+  return (
+    <NetworkContainer>
+      <PanelActions>
+        <PageSubtitle>{intl.formatMessage({ id: 'planes' })}</PageSubtitle>
```
We should not translate this `planes`.

ChengYanJin commented on ui/src/components/DashboardPlane.test.js:
```diff
@@ -0,0 +1,96 @@
+const NB_ITEMS = 2;
+
+describe("the dashboard network's plane panel", () => {
+  test("displays the network's plane panel and display 2 green statuses when no alerts are present", async () => {
```
```suggestion
  test("displays 2 green statuses when no alerts are present", async () => {
```

ChengYanJin reviewed (commented):
Apart from the comment, I think we should have `StartsAt` field in the tooltip when the status is NOT healthy. 

I am not sure if we should display the arrow, I will let @Cuervino answer the question ;)

ChengYanJin commented on ui/src/components/DashboardPlane.test.js:
```diff
@@ -39,17 +39,34 @@ jest.mock('../containers/AlertProvider', () => ({
   },
 }));
 
+jest.mock('../containers/ConfigProvider', () => ({
```
I don't think the work is related to the commit message :(

And please refer to the best practice of commit messages. 

https://metal-k8s.readthedocs.io/en/latest/developer/development/best_practices/commit.html?highlight=commit#subject

ChengYanJin commented on ui/src/components/HealthItem.js:
```diff
@@ -132,15 +143,21 @@ const HealthItem = ({
-          <Link to="/alerts" data-testid="alert-link">
+          <div
+            onClick={() => {
+              history.push('/alerts');
```
could be removed, the openLink() already take care of this. 
```suggestion

```

JulienMeziere replied:
With the openLink() only, the "alert" page gets selected in the navbar but the page doesn't show up.
Before what we did, a Link component was used and it was showing the alert page but not selecting "alert" in the navbar.
Is there something I am missing to navigate using openLink ?

ChengYanJin replied:
Please use `history.replace` instead of `history.push`

And clean up the commits with the best practice of the commit message. 

Then we are good to go.

Thanks!

JBWatenbergScality replied:
To be precise the `history.replace` should be called after `openLink`.
This is necessary because openLink triggers a change on `shell-ui` history which renders the same federated component coming from Metalk8s (/ is matched for /alerts) then Metalk8s federated UI doesn't detect any changes in it's history and doesn't rerender.
Using `history.replace` before calling openLink would replace the current browsed page history. Calling it after would replace the change performed by `openLink`, so you should be browsed to the Alert page and pressing once the back button should bring you back to the overview page.

ChengYanJin commented on ui/src/components/HealthItem.js:
```diff
@@ -0,0 +1,166 @@
+          // Right placement to avoid Z index issues between left sidebar and tooltip or out of screen tooltip displays
+          placement="top"
+          overlayStyle={{
+            // em sizing to handle font-size change on large screens
```
the comment could be removed
```suggestion
```

ChengYanJin commented on ui/src/components/DashboardPlane.js:
```diff
@@ -44,13 +43,15 @@ const DashboardPlane = () => {
             label={intl.formatMessage({ id: 'control_plane' })}
             status={planesStatus}
             alerts={planesHighestSecurityAlert}
+            showArrow={false}
```
I checked with @Cuervino, so I think you were right, let's remove the arrow here.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @lucieleonard

* @teddyandrieux

* @thomasdanan

* @nicolast

* @chengyanjin

* @jbwatenbergscality

* @gdemonet

* @alexandre-allard-scality





The following reviewers are expecting changes from the author, or must review again:

* @chengyanjin

ChengYanJin commented on ui/webpack.common.js:
```diff
@@ -29,7 +29,7 @@ module.exports = (env) => ({
     // ~732 KiB for production
     // ~4.06 MiB for development because flow increase the size of assets.
-    maxAssetSize: process.env.NODE_ENV === 'production' ? 750000 : 5000000,
+    maxAssetSize: process.env.NODE_ENV === 'production' ? 750000 : 100000000,
```
I suggest dropping this change because we have already handled it on our side.

ChengYanJin replied:
and you may want to fixup this commit with the first one

Commit:
ui/dashboard: Add network plane health in Network panel

Remove the ServiceItem from DashboardServices and create the HealthItem component with it
Add DashboardPlane component containing 2 HealthItems
Add tests for the new DashbordPlane component
history.push replaced by history.replace in the HealthItem component

Ref: #3511

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @lucieleonard

* @teddyandrieux

* @thomasdanan

* @nicolast

* @chengyanjin

* @jbwatenbergscality

* @gdemonet

* @alexandre-allard-scality

JBWatenbergScality commented:
Cherry-picked in #3501
